src/kibana/components/vislib/lib/legend.js

```diff
@@ -37,6 +43,70 @@ define(function (require) {
     }
 
     /**
+     * Returns an arr of data objects that includes labels, aggConfig, and an array of data values
+     * for the pie chart.
+     */
+    Legend.prototype._transformPieData = function (arr) {
+      var data = [];
+      var recurse = function (obj) {
+        if (obj.children) {
+          recurse(obj.children).reverse().forEach(function (d) { data.unshift(d); });
+        }
+        return obj;
+      };
+
+      arr.forEach(function (chart) {
+        chart.slices.children.map(recurse)
+        .reverse().forEach(function (d) { data.unshift(d); });
+      });
+
+      return _.unique(data, function (d) { return d.name; });
+    };
+
+    /**
+     * Filter out zero injected objects
+     */
+    Legend.prototype._filterZeroInjectedValues = function (arr) {
+      return arr.filter(function (d) {
+        return d.aggConfigResult !== undefined;
+      });
+    };
+
+    /**
+     * Returns an arr of data objects that includes labels, aggConfig, and an array of data values
+     * for the point series charts.
+     */
+    Legend.prototype._transformSeriesData = function (arr) {
+      var data = [];
+      var self = this;
+
+      arr.forEach(function (chart) {
+        chart.series.forEach(function (obj, i) {
+          var currentObj = data[i];
+          if (!currentObj) data.push(obj);
+
+          // Copies first aggConfigResults object to data object.
+          if (obj.values && obj.values.length) {
+            var values = self._filterZeroInjectedValues(obj.values);
+            var aggConfigResult = values.length && values[0].aggConfigResult ?
+              values[0].aggConfigResult.$parent : undefined;
```

Comment:
Can you can you change this to:

```javascript
var aggConfigResult;
if (values.length && values[0].aggConfigResult) {
  aggConfigResult = values[0].aggConfigResult.$parent;
}
```

Multiline ternaries are against our style guide